napari/layers/base/base.py

```diff
@@ -458,10 +461,15 @@ def _mode_setter_helper(self, mode: Union[Mode, str]) -> Mode:
             New mode for the current layer.
 
         """
-        mode = self._modeclass(mode)
+        mode = self._modeclass(mode_in)
+        # Sub-classes can have their own Mode enum, so need to get members
+        # from the specific mode class set on this layer.
+        PAN_ZOOM = self._modeclass.PAN_ZOOM
+        TRANSFORM = self._modeclass.TRANSFORM
```

Comment:
```suggestion
        PAN_ZOOM = self._modeclass.PAN_ZOOM # type: ignore[attr-defined]
        TRANSFORM = self._modeclass.TRANSFORM # type: ignore[attr-defined]
```

I think that was the reason for the getattr/assert None. Let's try with type ignore. 
TO properly fix that we'll need Generics over Modeclass, and we are not there yet.